src/backend/distributed/planner/multi_planner_utils.c

```diff
@@ -0,0 +1,863 @@
+/*
+ * multi_planner_utils.c
```

Comment:
Still disagree with the file name - sounds like minor helpers, rather than some fundamental infrastructure.